src/biogeophys/SoilStateInitTimeConstMod.F90

```diff
@@ -303,22 +302,19 @@ subroutine SoilStateInitTimeConst(bounds, soilstate_inst, nlfilename)
     ! get original soil depths to be used in interpolation of sand and clay
     ! --------------------------------------------------------------------
 
-    allocate(zsoifl(1:nlevsoifl), zisoifl(0:nlevsoifl), dzsoifl(1:nlevsoifl))
-    do j = 1, nlevsoifl
-       zsoifl(j) = 0.025*(exp(0.5_r8*(j-0.5_r8))-1._r8)    !node depths
-    enddo
-
-    dzsoifl(1) = 0.5_r8*(zsoifl(1)+zsoifl(2))             !thickness b/n two interfaces
-    do j = 2,nlevsoifl-1
-       dzsoifl(j)= 0.5_r8*(zsoifl(j+1)-zsoifl(j-1))
+    ! Note that the depths on the file are assumed to be the same as the depths in the
+    ! model when running with 10SL_3.5m. We need to specify zsoifl down to nlevsoifl+1
+    ! (rather than just nlevsoifl) so that we can get the appropriate zisoifl at level
+    ! nlevsoifl (i.e., the bottom interface depth).
+    allocate(zsoifl(1:nlevsoifl+1), zisoifl(0:nlevsoifl))
+    do j = 1, nlevsoifl+1
+       zsoifl(j) = 0.025_r8*(exp(0.5_r8*(j-0.5_r8))-1._r8)    !node depths
     enddo
-    dzsoifl(nlevsoifl) = zsoifl(nlevsoifl)-zsoifl(nlevsoifl-1)
 
     zisoifl(0) = 0._r8
-    do j = 1, nlevsoifl-1
+    do j = 1, nlevsoifl
```

Comment:
Diff (3) from my main PR comment (and related changes around here): changing definition of zisoifl for lowest level

Reply:
I agree that this feels weird, and certainly I'd feel less comfortable with this if zsoifl were used for anything more than defining zisoifl. But since zsoifl is just used to define zisoifl, I figured that the best thing to do here is to define zsoifl so that the zisoifl values end up as desired - which involves setting an assumed value for zsoifl(nlevsoifl+1). Unless you see a cleaner way to accomplish this goal?

Reply:
Sounds good. To help with the long term development and institutional memory of what is going on here, how about a comment that explains how this code might change if we **were** reading the depths from the file (a file).; and which **specific** file we are referring to; and the danger of switching to a new file given the current implementation. If such a comment exists somewhere else, I suggest copying it here, too.

Reply:
Comment added in 40a5881c